language/stdlib/modules/libra_account.move

```diff
@@ -409,70 +458,131 @@ module LibraAccount {
                 authentication_key,
                 delegated_key_rotation_capability: false,
                 delegated_withdrawal_capability: false,
-                received_events: Event::new_event_handle_from_generator<ReceivedPaymentEvent>(&mut generator),
-                sent_events: Event::new_event_handle_from_generator<SentPaymentEvent>(&mut generator),
+                received_events: Event::new_event_handle<ReceivedPaymentEvent>(&new_account),
+                sent_events: Event::new_event_handle<SentPaymentEvent>(&new_account),
                 sequence_number: 0,
                 is_frozen: false,
-            },
-            generator,
-            fresh_address,
+            }
         );
+        // (4) publish Account::Role. it's the caller's job to publish resources with role-specific
+        //     configuration such as AccountLimits before calling this
+        move_to(&new_account, Role<RoleData> { role_data });
+        // (5) publish Balance resource(s)
+        move_to(&new_account, Balance<Token>{ coin: Libra::zero<Token>() });
+        destroy_signer(new_account);
+    }
+
+    /// Create an account with the Empty role at `new_account_address` with authentication key
+    /// `auth_key_prefix` | `new_account_address`
+    // TODO: needed?
+    public fun create_empty_account<Token>(
+        new_account_address: address,
+        auth_key_prefix: vector<u8>
+    ) {
+        let new_account = create_signer(new_account_address);
+        AccountLimits::publish(&new_account);
+        make_account<Token, Empty::T>(new_account, auth_key_prefix, Empty::create())
+    }
+
+    /// Create a treasury/compliance account at `new_account_address` with authentication key
+    /// `auth_key_prefix` | `new_account_address`
+    public fun create_treasury_compliance_account<Token>(
+        new_account_address: address,
+        auth_key_prefix: vector<u8>,
+    ) {
+        Association::assert_sender_is_root();
+        let new_account = create_signer(new_account_address);
+        Association::grant_association_address(&new_account);
+        Association::grant_privilege<FreezingPrivilege>(&new_account);
+        Libra::publish_burn_capability<Coin1::T>(&new_account);
+        Libra::publish_burn_capability<Coin2::T>(&new_account);
+        Libra::publish_burn_capability<LBR::T>(&new_account);
+        // TODO: add Association or TreasuryCompliance role instead of using Empty?
+        make_account<Token, Empty::T>(new_account, auth_key_prefix, Empty::create())
+    }
+
+    /// Create an account with the ParentVASP role at `new_account_address` with authentication key
+    /// `auth_key_prefix` | `new_account_address`.
+    public fun create_parent_vasp_account<Token>(
+        new_account_address: address,
+        auth_key_prefix: vector<u8>,
+        human_name: vector<u8>,
+        base_url: vector<u8>,
+        compliance_public_key: vector<u8>,
+    ) {
+        Association::assert_sender_is_association();
```

Comment:
We can do this in a different PR, but do we want this to be the root account? Or some specific privilege?

Reply:
Good question and I'm not sure (probably the TC account?). Will add a TODO